lib/straight/gateway.rb

```diff
@@ -90,32 +89,34 @@ def new_order(args)
         if args[:amount].nil? || !args[:amount].kind_of?(Numeric) || args[:amount] <= 0
           raise OrderAmountInvalid, "amount cannot be nil and should be more than 0"
         end
-        # Setting default values
-        args[:currency]         ||= default_currency
-        args[:btc_denomination] ||= :satoshi
 
-        amount = args[:amount_from_exchange_rate] = amount_from_exchange_rate(
+        # Build order with default values
+        order = Kernel.const_get(order_class).new
+        order.gateway = self
+        order.keychain_id = args[:keychain_id]
+        order.currency = args[:currency] || default_currency
+        order.block_height_created_at = fetch_latest_block_height rescue nil
+
+        args[:amount_from_exchange_rate] = amount_from_exchange_rate(
           args[:amount],
-          currency:         args[:currency],
-          btc_denomination: args[:btc_denomination]
+          currency: order.currency,
+          btc_denomination: args[:btc_denomination] || :satoshi,
         )
 
+        order.amount = args[:amount_from_exchange_rate]
+
         if address_provider.takes_fees?
-          address, amount = address_provider.new_address_and_amount(**args)
+          order.address, order.amount =
+            address_provider.new_address_and_amount(**args)
         else
-          address = address_provider.new_address(**args)
+          order.address = address_provider.new_address(**args)
         end
 
-        order             = Kernel.const_get(order_class).new
-        order.gateway     = self
-        order.keychain_id = args[:keychain_id]
-        order.address     = address
-        order.amount      = amount
-        order.block_height_created_at = fetch_latest_block_height rescue nil
-
-        unless args[:currency] == "BTC"
-          order.exchange_rate = current_exchange_rate(args[:currency])
-          order.currency = args[:currency]
+        if address_provider.takes_fees? && order.currency != "BTC"
+          order.amount_with_currency =
+            format("%.2f %s", args[:amount], order.currency)
+        elsif order.currency != "BTC"
+          order.exchange_rate = current_exchange_rate(order.currency)
```

Comment:
In widget-app we have 
```
if object.exchange_rate
  format("%.2f %s", object.exchange_rate * amount_in_btc, object.currency)
...
```
So why do we need to do this calculation? I think you can leave just 
```
unless order.currency == 'BTC'
  order.amount_with_currency = format("%.2f %s", args[:amount], order.currency)
end
```
and remove `exchange_rate `, `currency ` fields from `straight-server`